client/signup/steps/choose-service/index.tsx

```diff
@@ -0,0 +1,143 @@
+
+const Placeholder = () => <span className="choose-service__placeholder">&nbsp;</span>;
+
+export default function ChooseServiceStep( props: Props ): React.ReactNode {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const displayCost = useSelector( ( state ) => getProductDisplayCost( state, WPCOM_DIFM_LITE ) );
+	const isLoading = useSelector( isProductsListFetching );
+
+	const headerText = translate( 'Let our experts create your dream site' );
+
+	const subHeaderText = translate(
+		'Get your business up quickly with our experts building your responsive, professional website. Select your best option.'
+	);
+
+	const intents: DIFMOrBuiltByIntent[] = [
+		{
+			key: 'difm',
+			title: translate( 'Do It For Me' ),
+			description: translate(
+				'Get a professionally designed, mobile-optimized website in %(fulfillmentDays)d business days or less for a one-time fee of {{PriceWrapper}}%(displayCost)s{{/PriceWrapper}} plus a one year subscription of the %(plan)s plan.',
+				{
+					args: {
+						displayCost,
+						fulfillmentDays: 4,
+						plan: isEnabled( 'plans/pro-plan' )
+							? getPlan( PLAN_WPCOM_PRO )?.getTitle()
+							: getPlan( PLAN_PREMIUM )?.getTitle(),
+					},
+					components: {
+						PriceWrapper: isLoading ? <Placeholder /> : <strong />,
+					},
+				}
+			),
+			icon: mouse,
+			value: 'difm',
+			actionText: translate( 'Get Started' ),
+			isPrimary: true,
+		},
+		{
+			key: 'builtby',
+			title: translate( 'Built by WordPress.com - Concierge' ),
+			description: translate(
+				'Perfect for those looking for advanced features (e.g., e-commerce, migrations), customized designs, or who prefer a custom, consultative experience.'
+			),
+			icon: headset,
+			value: 'builtby',
+			actionText: translate( 'Learn More' ),
+		},
+	];
+
+	useEffect( () => {
+		if ( ! props.queryObject.siteSlug && ! props.queryObject.siteId ) {
+			throw new Error(
+				'website-design-services did not provide the site slug or site id it is configured to.'
+			);
+		}
+		dispatch( saveSignupStep( { stepName: props.stepName } ) );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
+
+	const onSelect = ( value: ChoiceType ) => {
+		if ( 'builtby' === value ) {
+			window.location.href = 'https://builtbywp.com/';
+			return;
+		}
+		dispatch(
+			submitSignupStep(
+				{ stepName: props.stepName },
+				{
+					siteSlug: props.queryObject.siteSlug || props.queryObject.siteId,
+				}
+			)
+		);
+		props.goToNextStep();
+	};
+
+	return (
+		<>
+			<QueryProductsList persist />
+			<StepWrapper
+				headerText={ headerText }
+				fallbackHeaderText={ headerText }
+				subHeaderText={ subHeaderText }
+				fallbackSubHeaderText={ subHeaderText }
+				stepContent={
+					<IntentScreen
+						intents={ intents }
+						onSelect={ onSelect }
+						preventWidows={ preventWidows }
+						intentsAlt={ [] }
+					/>
+				}
+				backUrl={ `/setup/intent?siteSlug=${ props.queryObject.siteSlug }` }
```

Comment:
Can we use something like the [addQueryArgs](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/url/add-query-args.ts) lib for this? 🤔

Reply:
I thought about this too, but I could not see any benefit of using this function instead of using the more straightforward string template. What do you think?